### DISCUSSION TITLE
function http_header(unknown, unknown) does not exist

### DISCUSSION THREAD

**Discussion by JakovGlavac** [supabase/supabase#29186] [Questions]
Hi,
i'm trying to create a simple function that will call post request. But i'm getting an error `function http_header(unknown, unknown) does not exist`. I've found a post fixing [simular issuse](https://github.com/orgs/supabase/discussions/19801?sort=top) but it didnt help. I dont know if i'm doing somthing wrong?

```sql
CREATE OR REPLACE FUNCTION send_new_user_event()
RETURNS trigger AS $$
BEGIN

    SET search_path TO public, extensions; 

    SELECT *
    FROM http((
      'POST',
      'https://api.june.so/sdk/track',
      ARRAY[http_header('Authorization', 'Basic 123124')],
      'application/json',
      jsonb_build_object(           
        'userId', '12345',           
        'event', 'Signup',             
        'properties', jsonb_build_object(
            'raw_user_meta_data', '{"country": "US", "age": 25}'::jsonb 
        )
      )::text                       
    ));
    RETURN NEW;  
END;
$$ LANGUAGE plpgsql;
```

**Comment by GaryAustin1** [comment]
Did you install the http extension into the extensions schema?

Also normally the search path goes in the function parameters definition section like:
```sql
$$  language plpgsql 
     set search_path = extensions, storage, pg_temp;
 ```
 But it may work in the plpgsql code, I've just not seen or done that.

**Reply by JakovGlavac** [reply]
I dont get where should i put search_path excacly in my query?

**Reply by GaryAustin1** [reply]
```
    ));
    RETURN NEW;  
END;
$$ LANGUAGE plpgsql set search_path = extensions,public;
```
As I said  I do not know that you can't do it your way.  The way I show (or in the beginning part of the function before $$) are the ways I've seen.

I take you have the extension enabled in extensions?

**Reply by GaryAustin1** [reply]
This is the code I use:
```sql
            select status from
                http((
                      'DELETE',
                      instance_url || '/storage/v1/object/' || bucket,
                      ARRAY[http_header('authorization','Bearer ' || service_role_key)],
                      'application/json',
                      file_body
                    )::http_request) into delete_status;  --not sure delete status is useful from storage-API in this case
            end if;
```
From https://github.com/GaryAustin1/supa-file-helper/blob/main/main.sql

**Reply by gevera** [reply]
> This is the code I use:
> 
> ```sql
>             select status from
>                 http((
>                       'DELETE',
>                       instance_url || '/storage/v1/object/' || bucket,
>                       ARRAY[http_header('authorization','Bearer ' || service_role_key)],
>                       'application/json',
>                       file_body
>                     )::http_request) into delete_status;  --not sure delete status is useful from storage-API in this case
>             end if;
> ```
> 
> From https://github.com/GaryAustin1/supa-file-helper/blob/main/main.sql

God bless you!

**Comment by DenisBessa** [comment]
This is the correct way:

```
select
    *
from
    extensions.http (
        (
            'POST',
            'https://your-url',
            ARRAY[
                extensions.http_header (
                    'Authorization',
                    '123'
                )
            ],
            'application/json',
            'select 1'
        )
    )
```

**Comment by HadiAbu** [comment]
You can use webhooks to call http request very easily without getting involved in writing code:

From the sidebar go to integrations and look for 'Database Webhooks' . After you install, very east to create a new HTTP request and add triggers from the UI

![image](https://github.com/user-attachments/assets/369832e9-87f4-4256-9dec-04ac19aec318)
![image](https://github.com/user-attachments/assets/2d30a2e5-b230-448e-9d24-171b86b50817)
